src/App.tsx

```diff
@@ -156,10 +181,31 @@ function APP() {
     );
   }
 
-  //states
-  const [treatment_units, update_treatment_units] = useState<string[]>([]);
-  const [selected_year, update_selected_year] = useState(2019);
-  const [selected_row, update_selected_row] = useState(null);
+  const [treatment_units, update_treatment_units] = useQueryParam(
+    "tu_names",
+    mainQueryParamsConfig.tu_names
+  );
+  const validated_treatment_units = validate_treatment_units(
+    treatment_units as string[],
+    tu_names
+  );
+  JSON.stringify(treatment_units) !==
+    JSON.stringify(validated_treatment_units) &&
+    treatment_units &&
+    update_treatment_units(validated_treatment_units);
```

Comment:
I see your point, this was done with intent as some props derived from query params actually break the app. It's ment more as a sanitiser than a type checker. Since we replaced state, query params are sent "blindly" down the line. We can remove the validators, but then we will have no sanitation what is delivered to  other components.

Also I'm not sure the use-case is to manipulating the url to achieve filtered results, I based it more on a point and click then share the url use-case. You can of course manipulate the url directly, but that will require "inside" knowledge of the params.

I suggest we leave it as is until we have refactored more of the app.

Reply:
Params are validated before they are handed down, I have removed the param type checker.. :)